Sources/KlaviyoSwift/StateManagement/StateManagement.swift

```diff
@@ -585,6 +588,16 @@ extension Store where State == KlaviyoState, Action == KlaviyoAction {
         reducer: KlaviyoReducer())
 }
 
+enum KlaviyoBadgeCountUtil {
+    static func syncBadgeCount() {
+        DispatchQueue.main.async {
```

Comment:
This feels side effect-y to me.

Reply:
felt safer this way since we are reading `UIApplication.shared.applicationIconBadgeNumber`

Reply:
Safer in what sense?

Reply:
thought calling UIApplication had to be called on main thread or is that an outdated/misremembered best practice